legion_core/src/world.rs

```diff
@@ -731,6 +786,33 @@ impl World {
     /// Determines if the given `Entity` is alive within this `World`.
     pub fn is_alive(&self, entity: Entity) -> bool { self.entity_allocator.is_alive(entity) }
 
+    /// Returns the entity's component types, if the entity exists.
+    pub fn entity_component_types(
```

Comment:
These functions would be very helpful. In my application, I also wanted to query components by an entity. With the current query system, I would have to use `iter_entities_mut` and then do an if check based on my entity and the iter result. I would love such a system. 

Though again, this will introduce a second way to query for components, is this wishable?

Reply:
I think this was added for handling serialization (and it could probably be done in a separate PR.) This is for querying what components are on the entity, not getting the component data itself. If there's a different way of doing this then maybe we don't need it.

Reply:
This is now in a separate PR